### DISCUSSION TITLE
Setting A Live User Password during Building the distribution images

### DISCUSSION THREAD

**Discussion by ghost** [SpiralLinux/SpiralLinux-project#342] [Q&A] [answered]
### Discussed in https://github.com/orgs/SpiralLinux/discussions/228

<div type='discussions-op-text'>

<sup>Originally posted by **bliu13** January 29, 2023</sup>
... didn't find any documentation for how to set up the environment or how to run the scripts...</div>

I have everything working for a successful build (which is great considering I have MS ...)
but for the life of me I cannot set  live user password ... I do not really understand how the debian live iso builder is supposed to set it via hooks?

I have been trying make my mother (who is 80) an enviroment that she cant mess with so she can go online to check her AOL email maybe play local games and stuff which is fine except I want a password set so she cannot use it whenever she wants.  She is a terrible tinkerer and will screw up a system if you let her.

She tries to use her cell phone as if it was a TV remote :(   SpiralLinux has been the closest I have gotten to having an enviroment she has a hard time really screwing with.  It doesnt have games but I am slowly fixing that by adding game packages and things BUT this is all pointless if it cannot be locked.

I know I must sound heartless but try supporting someone prown to deleting or changing the system :/   I have worked in IT 40 years but I am disabled now and I have issues understanding things like how to set the live user password?

I would rather not have to keep installing but I will - then I would try penguin-eggs to see if I can make an ISO to be loadable by Ventoy - that way it is pretty much readonly and gets reset with every reboot.

So, my question is - what is the proper Hook Format or Syntax so I can set the Live User acct password?

Thx/Bill

**Comment by ghost** [comment]
Ok, OMG, I got the live user password set but it set to live ... which I have no idea how I did that  :(

I have done 5 builds already today and it is just now  6am my time.  I guess 'live' is better than being reset to nothing  ... 

Also, where does the Calamarie installer get pickup.  In this implementation it isnt needed cuz this ISO will never get installed.  I had thought I removed it but it wasnt the right package ... system still runs without whatever package I deleted  ><

Thx/Bill

**Comment by geckolinux** [comment, marked as answer]
Hi there, there are several ways to accomplish this, but the easiest would probably be by modifying `config/includes.chroot/lib/live/config/2000-spirallinux` . The file is commented, so you'll see which sections to modify or eliminate.

**Reply by ghost** [reply]
Thank you very much  :)

I was able to use:
echo "user:userpw" | chpasswd

(which I thought about a couple days ago but it wasnt sticking  :(

Thx/Bill :)
